internal/cache/file/cache_handler.go

```diff
@@ -56,16 +56,30 @@ type CacheHandler struct {
 
 	// excludeRegex is the compiled regex for excluding files from cache
 	excludeRegex *regexp.Regexp
+
+	// includeRegex is the compiled regex for including files from cache
+	includeRegex *regexp.Regexp
 }
 
-func NewCacheHandler(fileInfoCache *lru.Cache, jobManager *downloader.JobManager, cacheDir string, filePerm os.FileMode, dirPerm os.FileMode, excludeRegex string) *CacheHandler {
-	var compiledRegex *regexp.Regexp
+func NewCacheHandler(fileInfoCache *lru.Cache, jobManager *downloader.JobManager, cacheDir string, filePerm os.FileMode, dirPerm os.FileMode, excludeRegex string, includeRegex string) *CacheHandler {
+	var compiledExcludeRegex *regexp.Regexp
+	var compiledIncludeRegex *regexp.Regexp
+
 	if excludeRegex != "" {
 		var err error
-		compiledRegex, err = regexp.Compile(excludeRegex)
+		compiledExcludeRegex, err = regexp.Compile(excludeRegex)
 		if err != nil {
 			logger.Warnf("Failed to compile exclude regex %q: %v", excludeRegex, err)
-			compiledRegex = nil
+			compiledExcludeRegex = nil
+		}
+	}
+
+	if includeRegex != "" {
+		var err error
+		compiledIncludeRegex, err = regexp.Compile(includeRegex)
+		if err != nil {
+			logger.Warnf("Failed to compile include regex %q: %v", includeRegex, err)
```

Comment:
we should exact this logic into a seperate method and use directly in line 93 & 94.